schism/mplink.c

```diff
@@ -222,6 +222,29 @@ int song_find_last_channel(void)
 
 // ------------------------------------------------------------------------
 
+int song_get_pattern_offset(int * n, song_note_t ** buf, int * r, int offset)
+{
+	int len, tot;
+	if (song_get_mode() & (MODE_PATTERN_LOOP)) {
+		// just wrap around current rows
+		*r = (*r + offset) % song_get_rows_in_pattern(*n);
+		return song_get_pattern(*n, buf);
+	}
+
+	tot = song_get_rows_in_pattern(*n);
+	while (offset + *r > tot) {
+		offset - tot;
```

Comment:
This statement doesn't do anything.

Reply:
correct. i meant `offset -= tot` fixing.